src/core.c/Main.pm6

```diff
@@ -51,10 +51,60 @@ my sub RUN-MAIN(&main, $mainline, :$in-as-argsfiles) {
         my $positional := nqp::create(IterationBuffer);
         my %named;
 
+        my &coercer = &val;
+        if %sub-main-opts<coerce-allomorphs-to>:exists {
+            my $type := %sub-main-opts<coerce-allomorphs-to><>;
+            if $type =:= Numeric {
```

Comment:
Since this probably isn't very hot code, why not:
```perl6
if $type =:= Numeric|Int|Rat|Num|Complex|Str {
    &coercer = -> \value {
        (my \result := val(value)) ~~ Allomorph
            ?? result.$type
            !! result
    }
} else {
```

Reply:
Sadly, that doesn't work for `Numeric`.  And the check for:

     $type =:= Int|Rat|Num|Complex|Str

doesn't work, because that would compare with the instantiated Junction, **not** with its constituents.

Reply:
Huh, I hadn't realized that for `Numeric`. I guess you could do:
```perl6
if $type =:= Int or $type =:= Rat or $type =:= Num or $type =:= Complex or $type =:= Str {
    &coercer = -> \value {
        (my \result := val(value)) ~~ Allomorph
            ?? result.$type()
            !! result
    }
}
elsif $type =:= Numeric {
    &coercer = -> \value {
        (my \result := val(value)) ~~ Allomorph
            ?? result.Numeric
            !! result
    }
}
else {
```
but maybe it doesn't lose enough copy/pasting to make up for the extra complexity.

Reply:
Simplified further with https://github.com/rakudo/rakudo/commit/49eecd68c5